addons/account/models/reconciliation_widget.py

```diff
@@ -153,26 +161,46 @@ def get_bank_statement_line_data(self, st_line_ids, excluded_ids=None):
         return results
 
     @api.model
-    def get_bank_statement_data(self, bank_statement_ids):
+    def get_bank_statement_data(self, bank_statement_ids, search_str=False):
```

Comment:
what is this change? everything done in this function doesn't seem related to the task... can you explain?

Reply:
```python
def get_bank_statement_data(self, bank_statement_line_ids, search_str=False):
    bank_statements = self.env['account.bank.statement.line'].browse(bank_statement_line_ids).mapped('statement_id')
```
that would be much cleaner...